test/core/SedaCoreV1.test.ts

```diff
@@ -102,36 +102,77 @@ describe('SedaCoreV1', () => {
 
     const data = { requests, results, proofs, wallets, initialBatch, validatorProofs };
 
-    return { prover, core, data };
+    return { prover, core, feeManager, data };
   }
 
-  describe('getPendingRequests', () => {
-    it('returns correct requests with pagination', async () => {
+  describe('postRequest', () => {
+    it('allows duplicate request submissions but does not repost', async () => {
       const { core, data } = await loadFixture(deployCoreFixture);
 
-      for (const request of data.requests) {
-        await core.postRequest(request);
-      }
+      let requests = await core.getPendingRequests(0, 10);
+      expect(requests.length).to.equal(0);
 
-      const requests1 = await core.getPendingRequests(0, 2);
-      const requests2 = await core.getPendingRequests(2, 2);
+      const pr1 = await core.postRequest(data.requests[0]);
+      expect(pr1)
+        .to.emit(core, 'RequestPosted')
+        .withArgs(await deriveRequestId(data.requests[0]));
 
-      expect(requests1.length).to.equal(2);
-      expect(requests2.length).to.equal(2);
-      expect(requests1[0]).to.not.deep.equal(requests2[0]);
-      expect(requests1[1]).to.not.deep.equal(requests2[1]);
+      const pr2 = core.postRequest(data.requests[0]);
+
+      expect(pr2).not.to.emit(core, 'RequestPosted');
+
+      requests = await core.getPendingRequests(0, 10);
+      expect(requests.length).to.equal(1);
     });
 
-    it('returns zero requests if offset exceeds length', async () => {
+    it('prevents reposting a request after its result is posted', async () => {
```

Comment:
Same here, believe this is allowed on core contract.

Reply:
We agreed that we wanted to avoid recurring data request in this implementation to avoid edge cases with the solver. It is true that the protocol supports it.